services/UserSocialMediaService.ts

```diff
@@ -0,0 +1,58 @@
+import { Service } from 'typedi';
+import { EntityManager } from 'typeorm';
+import { InjectManager } from 'typeorm-typedi-extensions';
+import { ForbiddenError, NotFoundError } from 'routing-controllers';
+import { UserError } from '../utils/Errors';
+import { UserSocialMediaModel } from '../models/UserSocialMediaModel';
+import { UserModel } from '../models/UserModel';
+import Repositories, { TransactionsManager } from '../repositories';
+import { Uuid, PublicUserSocialMedia, SocialMedia } from '../types';
+
+@Service()
+export default class UserSocialMediaService {
+  private transactions: TransactionsManager;
+
+  constructor(@InjectManager() entityManager: EntityManager) {
+    this.transactions = new TransactionsManager(entityManager);
+  }
+
+  public async getSocialMediaForUser(user: UserModel): Promise<PublicUserSocialMedia[]> {
+    const userSocialMedia = await this.transactions.readOnly(async (txn) => Repositories.userSocialMedia(txn)
+      .getSocialMediaForUser(user));
+    return userSocialMedia.map((sm) => sm.getPublicSocialMedia());
+  }
```

Comment:
nit: Instead of returning the `Public` type in the service, let's return the model directly, and map it to its `Public` type in the API layer. This pattern is more ideal since the `Public` type is a concept specific to the API layer (since it defines the structure of the responses of the API to the client), so it should be used only in the API layer. We do still have remnants of services returning the `Public` type in other places in the code, but it's something we should move away from and eventually remove from the service layer entirely in a separate refactor.

Let's make this change for each of the functions in this service and their corresponding API layer equivalents.